cypress/support/PersistentLogs.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.  
+ * Licensed under the MIT License.
+ */
+
+var testCaseRegistry = [];
+var indexRegistry = 0;
+var logEntries = '';
+
+// Override the original console.log() function
+var originalConsolLog = console.log;
+
+// This becomes the effective console.log() function.
+console.log = function (message) {
+  originalConsolLog.apply(console, arguments);
+  logEntries += message + '\r\n';
+};
+
+// This will be called by the test list manager to register each test case
+// in the order the tests will be run in.
+Cypress.PersistentLogs = {};
+Cypress.PersistentLogs.RegisterTestCase = (testGroupName, testName) => {
+  testCaseRegistry.push(`${testGroupName}/${testName}`);
+}
+
+// After each test case ends...
+afterEach(() => {
+  var logFileName = `./results/cypress/${testCaseRegistry[indexRegistry++]}.${Cypress.moment().format("YY.MM.DD.HH.mm.ss..SSS")}.log`;
+  cy.writeFile(logFileName, logEntries);  
+  logEntries = '';
+});
```

Comment:
Does Cypress offer a async/non-blocking file operation. I ask given the suspected timing issues we've encountered in the past.

Reply:
I believe this is async because you can chain a .then() function off of it, they offer no other version. But you did bring up a good point, I should clear those logEntries in the chained .then() function...not even sure why it worked that way but it did.

I fixed the code.